### ISSUE TITLE
Use the classifier alone?

### ISSUE THREAD

**Issue by at-land** [fizyr/keras-retinanet#880]
Hi
I want to separate the classifier from the full model
Can I do that
Or at least can I pass the boxes directly to a classifier?
I know it's a little weird but I want to use the same classifier instead of waiting for long hours until the new training ends.
This will facilitate my trying.
I wish  guidance, thank you

**Comment by martinzlocha** [comment]
> I want to separate the classifier from the full model
> Can I do that

This is not really possible, you need the whole backbone for the classification part to work. So theoretically you could keep the whole backbone and just remove the regression part of the network but I am not sure if this would achieve your goal.

> Or at least can I pass the boxes directly to a classifier?

I would recommend that you read up on how RetinaNet works, it is a one-shot detector. This means that during inference you classify all of the bounding boxes over the whole image. So theoretically yes you can then only select the bounding boxes which you want but again you need to have the whole backbone.

> I know it's a little weird but I want to use the same classifier instead of waiting for long hours until the new training ends.

I'm not exactly sure what you are trying to achieve. However one thing that you could do is when starting training you freeze the backbone and the classification sub-models. This will essentially mean that only the regression sub-models will get trained if thats your goal. However freezing both backbone and the classification sub-models sounds very unusual so not sure how much it would help.

**Comment by at-land** [comment]
Thank you very much for all your very useful words.
I wanted to experiment with regression training with the size of the anchor in each experiment, without having to train the same classifier every time.
But experience says the performance of the classifier is very high, so I thought I could separate them.
The size of the boxes on which the classifier is trained may vary.
But it is clear that I need to re-read and understand the paper well.
I'am sorry my level is very poor, I will work to understand it.
Thank you so much for clarifying things
,
But please other question
Is there a way to train objects with rotation

Rather than horizontally and vertically only w and h
There are several points as polygon
y1, y2, y3, y4, x1, y2, x3, x4
Because it is my problem with the discovery of small elements with the possibility of rotation
I spent a lot of time with training using a larger anchor size.
Quality is not so high
Thanks a lot

**Comment by martinzlocha** [comment]
I think that training the regressor without the classifier is a very bad idea. The classifier is trained specifically for certain scales/ratios and you will probably get weird results if you make major changes to those. If you want to improve your training times I would suggest that you train a model on your dataset and then in the future you load the weights and freeze it. This will reduce the number of parameters which have to be trained.

Also I'm not sure if I understand your rotation question correctly. But yes, RetinaNet will by default transform your images. Eg rotate them etc. You just have ti configure that.

**Comment by at-land** [comment]
Hi
Thank you very much for your interest in reply
I'm grateful you have explained to me some things that I misunderstood.

Regarding rotation
I meant instead of getting a rigid box around the objects.
I want to make it like the following picture

![images](https://user-images.githubusercontent.com/47000745/51746332-73295e00-20ae-11e9-8b78-7e83a5713f80.jpg)
So that it recreates the shape of the slanted body as it is
Thank.

**Comment by martinzlocha** [comment]
Hmm well to do that you could change how the regression subnetwork works. Currently it returns just the top left and bottom right  coordinates which make up the bounding box. You could instead return all 4 corners of the box.

Another approach is to either use Mask R-CNN or Retina U-Net (https://arxiv.org/abs/1811.08661). Retina U-Net might give you better results as it also produces semantic segmentation masks however it will most probably be much more work.

**Comment by at-land** [comment]
Very cool
Well I started b already with {keras-maskrcnn}
I will complete your proposal. Retina U-Net
I'll include that in the title of the problem. Because it's a solution to another problem.
Thank you so much
